packages/react-events/src/__tests__/Swipe-test.internal.js

```diff
@@ -0,0 +1,326 @@
+      expect(directions).toEqual(['left']);
+    });
+    it('should be up', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, 50));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['up']);
+    });
+    it('should be down', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, -50));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['down']);
+    });
+
+    it('directions in events in the correct order vertical', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', -50, 0));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 50, 0));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 60, 0));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['left', 'right', 'right']);
+    });
+    it('directions in events in the correct order horizontal', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, -50));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, 50));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, 60));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['down', 'up', 'up']);
+    });
+  });
+
+  describe('nested responders', () => {
+    //TODO: test for many touch identifier
+    it('dispatch events in the correct order', () => {
+      let events = [];
+      const ref = React.createRef();
+      const createEventHandler = msg => () => {
+        events.push(msg);
+      };
+
+      const element = (
+        <Swipe
+          onSwipe={createEventHandler('outer: onSwipe')}
+          onSwipeEnd={createEventHandler('outer: onSwipeEnd')}
+          onSwipeStart={createEventHandler('outer: onSwipeStart')}>
+          <Swipe
+            onSwipe={createEventHandler('inner: onSwipe')}
+            onSwipeEnd={createEventHandler('inner: onSwipeEnd')}
+            onSwipeStart={createEventHandler('inner: onSwipeStart')}>
+            <div ref={ref} />
+          </Swipe>
+        </Swipe>
+      );
+
+      ReactDOM.render(element, container);
+
+      ref.current.dispatchEvent(createMouseEvent('pointerdown'));
+      ref.current.dispatchEvent(createMouseEvent('pointermove', 50, 50));
+      ref.current.dispatchEvent(createMouseEvent('pointerup'));
+      expect(events).toEqual([
+        'inner: onSwipeStart',
+        'outer: onSwipeStart',
+        'inner: onSwipe',
+        'outer: onSwipe',
+        'inner: onSwipeEnd',
+        'outer: onSwipeEnd',
+      ]);
+    });
+  });
+
+  describe('PointerType', () => {
```

Comment:
This isn't really a top-level test, as it can be folded into event object assertions for each of the callbacks